### DISCUSSION TITLE
Is there a way to stop traversal based on edge property ?

### DISCUSSION THREAD

**Discussion by vineeth-brt** [ArcadeData/arcadedb#998] [Q&A] [answered]
Hi

We have a requirement , we need to traverse from a node(vertex) and the traverse should not be continue whenever there is an edge with a specific property value (says deleted = true) .  We found a WHILE clause in traverse query , but no idea whether we can use this with edge property.  Can you please advice whether we can solve this use case using traverse method or we need to go for multiple queries. Thank you in advance.

Regards
Vineeth

**Comment by lvca** [comment]
Have you already looked at https://docs.arcadedb.com/#Deep-Traversal?

**Reply by vineeth-brt** [reply]
Thank you @lvca  for your quick response. We tried different combination but failed to get the exact results. I can explain the use case a further more, We have a graph with aggregation and association relations from a node, we need to delete a node which should in turn delete all the relations (including vertex) under that node if there is an aggregation relationship.  For example , in the below diagram - when we traverse from Node1 - it should return Node 1, Node 2 , Node 3 and Node 4. 
In Node3 , should stop the traversal as we have an association relation only from Node3 (The name of the association can be anything. The idea is, can make a parent edge type as Association and all association relation can be derived from Association type or  can keep a property isAssociation = true for association relations).

![graph](https://user-images.githubusercontent.com/104361357/229604483-c6169003-bdf8-49f6-86b5-75b312f2ac1a.png)

Meanwhile , we are trying to get more understanding on deep traversal based on the documentation.

Regards
Vineeth

**Reply by lvca** [reply, marked as answer]
Thanks for the explanation. Quick question, can aggregation be represented with inheritance in ArcadeDB? In this way, you'd have only 1 vertex and it would be much more efficient from all the standpoints. 

About the match, could you please write here your query that doesn't work?
